mephisto/abstractions/providers/prolific/prolific_datastore.py

```diff
@@ -0,0 +1,527 @@
+
+    def create_qualification_mapping(
+        self,
+        qualification_name: str,
+        requester_id: str,
+        prolific_project_id: str,
+        prolific_participant_group_name: str,
+        prolific_participant_group_id: str,
+    ) -> None:
+        """
+        Create a mapping between mephisto qualification name and Prolific
+        Participant Group details in the local datastore.
+
+        Repeat entries with the same `qualification_name` will be idempotent
+        """
+        try:
+            with self.table_access_condition, self._get_connection() as conn:
+                c = conn.cursor()
+                c.execute(
+                    """
+                    INSERT INTO qualifications(
+                        qualification_name,
+                        requester_id,
+                        prolific_project_id,
+                        prolific_participant_group_name,
+                        prolific_participant_group_id
+                    ) VALUES (?, ?, ?, ?);
+                    """,
+                    (
+                        qualification_name,
+                        requester_id,
+                        prolific_project_id,
+                        prolific_participant_group_name,
+                        prolific_participant_group_id,
+                    ),
+                )
+                return None
+
+        except sqlite3.IntegrityError as e:
+            if is_unique_failure(e):
+                # Ignore attempt to add another mapping for an existing key
+                db_qualification = self.get_qualification_mapping(qualification_name)
+
+                logger.debug(
+                    f'Multiple Prolific mapping creations '
+                    f'for qualification "{qualification_name}". '
+                    f'Found existing one: {db_qualification}. '
+                )
+                assert \
+                    db_qualification is not None, \
+                    'Cannot be none given is_unique_failure on insert'
+
+                db_requester_id = db_qualification['requester_id']
+                db_prolific_qualification_name = db_qualification['prolific_participant_group_name']
+
+                if db_requester_id != requester_id:
+                    logger.warning(
+                        f'Prolific Qualification mapping create for {qualification_name} '
+                        f'under requester {requester_id}, already exists under {db_requester_id}.'
+                    )
+
+                if db_prolific_qualification_name != prolific_participant_group_name:
+                    logger.warning(
+                        f'Prolific Qualification mapping create for {qualification_name} '
+                        f'with Prolific name {prolific_participant_group_name}, '
+                        f'already exists under {db_prolific_qualification_name}.'
+                    )
+
+                return None
+            else:
+                raise e
```

Comment:
We'll have to revisit these as part of the qualification implementation. Generally we'll have task-run <-> (allow participant group, deny participant group) mappings, and so these functions will be adapted for that.